lib/src/formatter.dart

```diff
@@ -39,25 +39,28 @@ class DetailedReporter extends SimpleFormatter {
   DetailedReporter(
       Iterable<AnalysisErrorInfo> errors, LintFilter filter, IOSink out,
       {int fileCount, String fileRoot, bool showStatistics: false,
-      quiet: false})
+      bool machineOutput: false, quiet: false})
```

Comment:
FWIW, when you get a large number of optional parameters, especially if you need to keep passing them all along (like here), I think it's worth considering defining an "options" object as an alternative. Your style might vary.